url27db8982cd_leohinterlang_argFace_tgz-pJ8-com_fidelis_argface_ArgSampleJ8/src/com/fidelis/argface/ArgHelp.java

```diff
@@ -0,0 +1,315 @@
+ *	1.0.0	Mar 18, 2014  Leo Hinterlang
+ *-
+ */
+package com.fidelis.argface;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * ArgFace Help Class.
+ * 
+ * @version 1.0.0
+ * @author Leo Hinterlang
+ *
+ */
+public class ArgHelp {
+    private ArgList         argList;
+    private List<ArgOption> optionList;
+    private boolean         suppressHelp;
+    private String          usageText;
+    private String          versionText;
+    private String          aboutText;
+    private String          helpText;
+    private List<ArgNode>   problemUsage = new ArrayList<ArgNode>();
+    private List<String>    problemText  = new ArrayList<String>();
+    
+    /**
+     * Sets the argument list.
+     * 
+     * @param argList the argument list
+     */
+    public void setArgList (ArgList argList) {
+        this.argList = argList;
+    }
+
+    /**
+     * Sets the option list.
+     * 
+     * @param optionList the option list
+     */
+    public void setOptionList (List<ArgOption> optionList) {
+        this.optionList = optionList;
+    }
+
+    /**
+     * Sets the "suppressHelp" operating mode.
+     * 
+     * @param suppressHelp {@code true} to suppress help
+     */
+    public void setSuppressHelp (boolean suppressHelp) {
+        this.suppressHelp = suppressHelp;
+    }
+
+    /**
+     * Sets the "version" text.
+     * 
+     * @param versionText the version text
+     */
+    public void setVersionText (String versionText) {
+        this.versionText = versionText;
+    }
+
+    /**
+     * Sets the "about" text.
+     * 
+     * @param aboutText the aboutText to set
+     */
+    public void setAboutText (String aboutText) {
+        this.aboutText = aboutText;
+    }
+
+    /**
+     * Sets the "help" text.
+     * 
+     * @param helpText the help text
+     */
+    public void setHelpText (String helpText) {
+        this.helpText = helpText;
+    }
+    
+    /**
+     * Prints the "usage" text.
+     * 
+     * @return {@code true} if the usage text was printed
+     */
+    public boolean printUsage () {
+        if (suppressHelp) {
+            return false;
+        }
+        return outputUsage();
+    }
+    
+    /**
+     * Outputs the "usage" text.
+     * 
+     * @return {@code true} if the usage text was output
+     */
+    public boolean outputUsage () {
+        if (usageText == null) {
+            usageText = buildUsageText();
+        }
+        if (usageText == null) {
+            return false;
+        }
+        System.out.println(usageText);
+        return true;
+    }
+    
+    /**
+     * Prints the "help" text.
+     * 
+     * @return {@code true} if the help text was printed
+     */
+    public boolean printHelp () {
+        if (suppressHelp) {
+            return false;
+        }
+        return outputHelp();
+    }
+    
+    /**
+     * Outputs the "help" text.
+     * 
+     * @return {@code true} if the help text was output
+     */
+    public boolean outputHelp () {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 24. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.